src/data/player.py

```diff
@@ -643,6 +528,318 @@ def set_board(self, board_id: int, board_number: int, color: BoardColor):
+                    opponent.rating_type,
+                )
+                for opponent in opponents
+            ]
+
+            # Minimum rating floor
+            if rating_list and rating_list[0].value < tn.minimum_rating:
+                rating_list[0].value = tn.minimum_rating
+                rating_list[0].type = PlayerRatingType.FIDE
+                res.adjusted_player = opponents[0]
+                res.adjusted_player_rating = tn.minimum_rating
+                rating_list.sort(key=attrgetter('value'))
+
+            res.num_rated_players = len(
+                [r for r in rating_list if r.type == PlayerRatingType.FIDE]
+            )
+
+            values = [r.value for r in rating_list]
+            avg = StaticUtils.round_ranking(sum(values) / len(values)) if values else 0
+            if avg < tn.minimum_average:
+                res.average_too_low = _(
+                    '<b>1.4.8a</b> The minimum average rating of the opponents for this norm is {min}.'
+                ).format(min=tn.minimum_average)
+
+            res.average_rating = avg
+
+            max_score = Result.WIN.points() * len(results_list)
+            bonus = StaticUtils.performance_bonus(score / max_score) if max_score else 0
+            performance = avg + bonus
+            res.performance = performance
+            draw_points = Result.DRAW.points()
+            if performance < tn.minimum_performance:
+                res.performance_too_low = _(
+                    '<b>1.4.8</b> The minimum performance for this norm is {min}.'
+                ).format(min=tn.minimum_performance)
+                new_score = score
+                new_bonus = bonus
+                draw_points = Result.DRAW.points()
+                while new_score < max_score:
+                    new_score += draw_points
+                    new_bonus = (
+                        StaticUtils.performance_bonus(new_score / max_score)
+                        if max_score
+                        else 0
+                    )
+                    if res.average_rating + new_bonus >= tn.minimum_performance:
+                        break
+                if res.average_rating + new_bonus >= tn.minimum_performance:
+                    res.performance_diff = score - new_score
+            else:
+                new_score = score
+                new_bonus = bonus
+                draw_points = Result.DRAW.points()
+                while new_score > 0:
+                    new_score -= draw_points
+                    new_bonus = (
+                        StaticUtils.performance_bonus(new_score / max_score)
+                        if max_score
+                        else 0
+                    )
+                    if res.average_rating + new_bonus < tn.minimum_performance:
+                        break
+                if res.average_rating + new_bonus < tn.minimum_performance:
+                    res.performance_diff = score - new_score
+
+        # 1.43d exception
```

Comment:
1.43 was the number before 2021
```suggestion
        # 1.4.3d exception
```